### PR TITLE
Added support for publishing NS records + official test vector 2 compliance

### PR DESCRIPTION
**Pull request by thehenrytsai**
1. Added support for publishing NS records and fixed bugs to achieve official test vector 2 compliance.
1. Removed `kid` in DNS records according to DID DHT spec update.
1. Minor renaming.
1. QoL - Updated CODEOWNERS to further increase review efficiency.
1. QoL - Added HTML code coverage output for `dids` repo for immediate coverage feedback.

### CODE REVIEW

File: packages/dids/.c8rc.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   ],
   "reporter": [
     "cobertura",
+    "html",
```
Comment (nitro-neal) [inline packages/dids/.c8rc.json:17]: did you mean for this to happen?

File: packages/dids/src/methods/did-dht.ts [hunk 2]
```diff
@@ -1175,13 +1179,13 @@ export class DidDhtDocument {
       const publicKeyBase64Url = Convert.uint8Array(publicKeyBytes).toBase64Url();
 
       // Define the data for the DNS TXT record.
-      const txtData = [`id=${methodId}`, `t=${keyType}`, `k=${publicKeyBase64Url}`];
+      const txtData = [`t=${keyType}`, `k=${publicKeyBase64Url}`];
```
Comment (nitro-neal) [inline packages/dids/src/methods/did-dht.ts:1182]: so id is out now because of updated spec?

File: packages/dids/src/methods/did-dht.ts [hunk 3]
```diff
@@ -1253,19 +1257,29 @@ export class DidDhtDocument {
...
       ttl  : DNS_RECORD_TTL,
       data : rootRecord.join(PROPERTY_SEPARATOR)
     });
 
+    // Add an NS record for each authoritative gateway URI.
+    for (const gatewayUri of authoritativeGatewayUris || []) {
+      nsRecords.push({
+        type : 'NS',
+        name : '_did.' + DidDhtDocument.getUniqueDidSuffix(didDocument.id) + '.', // name of an NS record a authoritative gateway MUST end in `<ID>.`
+        ttl  : DNS_RECORD_TTL,
```
Comment (nitro-neal) [inline packages/dids/src/methods/did-dht.ts:1272]: this is not configurable in the spec?

**Review by nitro-neal — APPROVED** [review]
dude great job!
One last piece of this is coming soon right? the `fromDnsPacket()` part?

File: packages/dids/.c8rc.json (hunk 1 above)
Comment (thehenrytsai) [reply packages/dids/.c8rc.json:17]: Yes: QoL - Added HTML code coverage output for `dids` repo for immediate coverage feedback.

File: packages/dids/src/methods/did-dht.ts (hunk 2 above)
Comment (thehenrytsai) [reply packages/dids/src/methods/did-dht.ts:1182]: Yes, see https://did-dht.com/#verification-methods.

File: packages/dids/src/methods/did-dht.ts (hunk 3 above)
Comment (thehenrytsai) [reply packages/dids/src/methods/did-dht.ts:1272]: It's configurable in the sense that it's an optional parameter for this method.

**Review by LiranCohen — APPROVED** [review]
🚀

**Review by csuwildcat — APPROVED** [review]

**Pull request merged**